solr/solrj/src/java/org/apache/solr/client/solrj/request/beans/RateLimiterPayload.java

```diff
@@ -71,4 +74,18 @@ public int hashCode() {
         slotAcquisitionTimeoutInMS,
         priorityBasedEnabled);
   }
+
+  public void maybeEnableForHost(String hostname) {
+    if (!this.enabled && !hostname.isEmpty()) {
+      if (this.nodesEnabled != null && !this.nodesEnabled.isEmpty()) {
```

Comment:
there's no reason not to have these all as one `if` clause. Fine this way too, just a little unusual.